lib/variable/include/scipp/variable/transform.h

```diff
@@ -379,10 +379,45 @@ template <class T> struct as_view {
 };
 template <class T> as_view(T &data, const Dimensions &dims) -> as_view<T>;
 
+template <class T>
+bool bad_variance_broadcast(const Dimensions &dims, const T &var) {
+  if (!var.has_variances())
+    return false;
+  // implicit broadcast
+  if (dims.ndim() > var.dims().ndim())
+    return true;
+  // there may be a stride==0 when an inner dim has length==0
+  if (dims.volume() == 0)
```

Comment:
Is this check needed? What kind of input has `volume == 0` but only non-zero strides?

Reply:
Yes it is needed. Without it, an input of `shape=(4,0)` is considered to broadcast variances.

Reply:
But doesn't this input have `stride[1]=0` if other inputs have `shape[1] != 0`? Otherwise, I don't see where it would broadcast.

Reply:
Exactly, it is not broadcasting, but without this check the function would return `True`.

Reply:
Ah, now I see!